components/Meme.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from "react";
+
+
+function Meme() {
+    const [meme, setMeme] = React.useState({
+        topText: "",
+        bottomText: "",
+        randomImage: "http://i.imgflip.com/1bij.jpg" 
+    })
+    const [allMemeImages, setAllMemeImages] = React.useState([])
+    console.log(allMemeImages)
+
+    useEffect(()=>{
+        fetch("https://api.imgflip.com/get_memes")
+        .then(resp=>resp.json())
+        .then(data=>setAllMemeImages(data.data.memes))
+    },[])
+    // console.log(allMemeImages)
+
+    // console.log(allMemeImages[3].url)
+    
+    function getMemeImage() {
+ 
+        const randomNumber = Math.floor(Math.random() * allMemeImages.length)
+        const url = allMemeImages[randomNumber].url
+        setMeme(prevMeme => ({
+            ...prevMeme,
+            randomImage: url
+        }))
+        
+    }
+    function getData(event){
+        setMeme(prevData=>{return{
+            ...meme,
+            [event.target.name]: event.target.value
+        }})
+    }
```

Comment:
- There is no need to use this return statement here, you can improve this to be like this: 
```suggestion
    function getData(event){
        setMeme(prevData=>({
            ...meme,
            [event.target.name]: event.target.value
        })
    }
```

- [This is optional ]: You can consider also first extracting the name and the values from e.target to make it look more clean like this: 

```suggestion
    function getData(event){
        const { name, value } = e.target;
        setMeme(prevData=>({
            ...meme,
            [name]: value
        })
    }
```